src/scene/renderer/forward-renderer.js

```diff
@@ -929,6 +929,12 @@ class ForwardRenderer {
 
             if (light.enabled) {
 
+                // make sure the lights update their final color if the scene
+                // use of physical units mismatch the lights internal flag
+                if (this.scene.physicalUnits !== light._usesPhysicalUnits) {
+                    light._updateFinalColor();
+                    light._usesPhysicalUnits = this.scene.physicalUnits;
+                }
```

Comment:
I would move this to a function on a Light class, for example setPhysicalUnits(boolean), and call it not from here, but few lines below from two places where we set `light.visibleThisFrame = true;` - so only for the visible lights.